### PR TITLE
Speed up build of sample folder

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Signed-off-by: Alan Jowett <alanjo@microsoft.com>

## Description

Build sample drivers and user mode components in parallel. 
Move each bpf2c build to its own folder.

## Testing

CI/CD

## Documentation

No.

### CODE REVIEW

**Review by gtrevi — DISMISSED** [review]

**Review by dthaler — APPROVED** [review]

**Review by saxena-anurag — DISMISSED** [review]

File: tests/sample/sample.vcxproj [hunk 1]
```diff
@@ -95,6 +95,8 @@
         popd
       </Command>
       <Outputs>$(OutputPath)%(Filename).o;$(OutputPath)%(Filename)_um.dll;$(OutputPath)%(Filename).sys</Outputs>
+      <!-- Don't run bpf2c in parrallel when built with fuzzing flags as this triggers failures. -->
```
Comment (dthaler) [inline tests/sample/sample.vcxproj:98]:
```suggestion
      <!-- Don't run bpf2c in parallel when built with fuzzing flags as this triggers failures. -->
```

File: tests/sample/sample.vcxproj [hunk 2]
```diff
@@ -110,6 +112,8 @@
         popd
       </Command>
       <Outputs>$(OutputPath)%(Filename).o</Outputs>
+      <!-- Don't run bpf2c in parrallel when built with fuzzing flags as this triggers failures. -->
```
Comment (dthaler) [inline tests/sample/sample.vcxproj:115]:
```suggestion
      <!-- Don't run bpf2c in parallel when built with fuzzing flags as this triggers failures. -->
```

File: tests/sample/sample.vcxproj [hunk 3]
```diff
@@ -133,6 +138,8 @@
         popd
       </Command>
       <Outputs>$(OutputPath)%(Filename).o;$(OutputPath)%(Filename)_um.dll;$(OutputPath)%(Filename).sys</Outputs>
+      <!-- Don't run bpf2c in parrallel when built with fuzzing flags as this triggers failures. -->
```
Comment (dthaler) [inline tests/sample/sample.vcxproj:141]:
```suggestion
      <!-- Don't run bpf2c in parallel when built with fuzzing flags as this triggers failures. -->
```

**Review by dthaler — APPROVED** [review]
Ok except for typos

**Review by dv-msft — DISMISSED** [review]

**Review by dv-msft — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]

**Pull request merged**